packages/manager/apps/catalog-revamp/src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react';
+import { OsdsButton } from '@ovhcloud/ods-components/button/react';
+import { OsdsText } from '@ovhcloud/ods-components/text/react';
+import { OsdsLink } from '@ovhcloud/ods-components/link/react';
+import { ODS_TEXT_LEVEL, ODS_TEXT_SIZE } from '@ovhcloud/ods-components/text';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import { useTranslation } from 'react-i18next';
+import FilterItem from './FilterItem';
+import './Filters.scss';
+import {
+  getAvailableCategoriesWithCounter,
+  getUniverses,
+  Product,
+} from '@/utils/utils';
+
+interface FiltersProps {
+  products: Product[];
+  setSelectedCategories: React.Dispatch<React.SetStateAction<string[]>>;
+  setSelectedUniverses: React.Dispatch<React.SetStateAction<string[]>>;
+  onApply: () => void;
+}
+
+const Filters: React.FC<FiltersProps> = React.memo(
+  ({ products, setSelectedCategories, setSelectedUniverses, onApply }) => {
+    const [selectedCategories, setSelectedCategoriesLocal] = useState<string[]>(
+      [],
+    );
+    const [selectedUniverses, setSelectedUniversesLocal] = useState<string[]>(
+      [],
+    );
+    const [hasInteracted, setHasInteracted] = useState<boolean>(false);
+
+    const setFilters = () => {
+      setSelectedCategories(selectedCategories);
+      setSelectedUniverses(selectedUniverses);
+      onApply();
+    };
+
+    // we get the list of available categories and universes from product list
+    const universes = getUniverses(products, true);
+    const categories = getAvailableCategoriesWithCounter(
+      products,
+      getUniverses(products, false),
+    );
+
+    const resetFilters = () => {
+      setSelectedCategoriesLocal([]);
+      setSelectedUniversesLocal([]);
+      setSelectedCategories([]);
+      setSelectedUniverses([]);
+    };
+
+    const { t } = useTranslation('catalog-revamp/filters');
+
+    const handleCheckboxChange = (
+      type: 'category' | 'universe',
+      label: string,
+      event: CustomEvent<{ checked: boolean; value: string }>,
+    ) => {
+      setHasInteracted(true);
+      const { checked } = event.detail;
+      if (type === 'category') {
```

Comment:
```suggestion
      const setState = type === 'category' ? setSelectedCategoriesLocal : setSelectedUniversesLocal;
```
Then you don't have to duplicate the function below

Reply:
I have tested it, the code seems much more short but it have side effect, when I close filters and re-open the checkbox is not checked anymore

Reply:
No problem of side effect with this refactor:
~~~
      const { checked } = event.detail;
      const updateState = (prevState: string[]) =>
        checked
          ? [...prevState, label]
          : prevState.filter((item) => item !== label);

      if (type === 'category') {
        setSelectedCategoriesLocal(updateState);
      } else {
        setSelectedUniversesLocal(updateState);
      }
~~~